src/tactic/apply_fun.lean

```diff
@@ -0,0 +1,93 @@
+import tactic.monotonicity
+import order.basic
+
+open tactic interactive (parse) interactive (loc.ns)
+     interactive.types (texpr location) lean.parser (tk)
+
+local postfix `?`:9001 := optional
+
+meta def apply_fun_name (e : expr) (h : name) (M : option pexpr) : tactic unit :=
+do {
+  H ← get_local h,
+  t ← infer_type H,
+  match t with
+  | `(%%l = %%r) := do
+      tactic.interactive.«have» h none ``(congr_arg %%e %%H),
```

Comment:
Is there a reason to use interactive `have` here instead of `note`? Not a big deal, but this doesn't seem idiomatic.

Reply:
The deep reason is that I copied from code I've seen. I'd be happy to replace that line with whatever feels idiomatic to you.